File: examples/python-model/models/order.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Optional

# <ai:customize hint="Add imports for Order field types">
# </ai:customize>


@dataclass
class Order:
    """Represents a order entity."""

    id: str
    title: str = None
    amount: Optional[Decimal] = None
    created_at: datetime = None

    # <ai:customize hint="Add domain methods for Order">
    def validate(self) -> bool:
        raise NotImplementedError
    # </ai:customize>

File: examples/rust-entity/src/entities/order.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Order {
    pub id: String,
    pub title: String,
    pub amount: Option<rust_decimal::Decimal>,
    pub shipped: bool,
}

impl Order {
    // <ai:customize hint="Add constructor and domain methods for Order">
    pub fn new(id: String) -> Self {
        todo!()
    }
    // </ai:customize>
}
